src/adv.py

```diff
@@ -52,63 +68,105 @@
 while True:
     print(new_player)
 
+
     print("Enter direction : ")
     print("North     : n")
     print("South     : s")
     print("East      : e")
     print("West      : w")
     print("To Quit   : q")
-
+    print("To pick up an item : take <item-name>")
+    print("To drop  an item : drop <item-name>")
 
     direction = input("Enter the new direction that you want to take :")
-    if direction not in ["n", "s", "e", "w", "q"]:
-        print("Enter valid direction")
-        continue
+    words = direction.strip().lower().split(" ")
 
-    #if direction is q, break
-    if direction =="q":
-        print("Game is Over!,Quit the game")
-        break
+    if len(words) == 1: #This is direction or 'q'
+        if direction not in ["n", "s", "e", "w", "q"]:
+            print("Enter valid direction")
+            continue
 
+        # if direction is q, break
+        if direction == "q":
+            print("Game is Over!,Quit the game")
+            break
+
+        # if direction == "n":
+        # check if room has north direction room.
+        # r is current room : payer.current_room
+        # if r.to_n is not None:
+        # move to r.to_n : player.current_room = r.to_n
+        # else:
+        #  you can't move
+
+        current_room = new_player.current_room
+        if direction == "n":
+            if current_room.n_to is None:  # if player's current room has no room in north direction
+                print("Can't move")
+                continue
+            else:
+                new_player.current_room = current_room.n_to
+
+        elif direction == "s":
+            if current_room.s_to is None:
+                print("Can't Move")
+                continue
+            else:
+                new_player.current_room = current_room.s_to
+
+        elif direction == "e":
+            if current_room.e_to is None:
+                print("Can't Move")
+                continue
+            else:
+                new_player.current_room = current_room.e_to
```

Comment:
Initially I wanted to make this code more DRY, but I can't think of a way to do it without making it more complicated in the process. I would say since there's only 4 directions here that this is totally acceptable